src/SFA.DAS.Tasks.Worker.UnitTests/MessageProcessors/ApprenticeUpdateCreatedMessageProcessorTests/WhenAApprenticeshipUpdateCreatedMessageIsProcessed.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.Commitments.Events;
+using SFA.DAS.EmployerAccounts.Events.Messages;
+using SFA.DAS.Messaging.Interfaces;
+using SFA.DAS.NLog.Logger;
+using SFA.DAS.Tasks.Application.Commands.SaveTask;
+using SFA.DAS.Tasks.API.Types.Enums;
+using SFA.DAS.Tasks.Worker.MessageProcessors;
+
+namespace SFA.DAS.Tasks.Worker.UnitTests.MessageProcessors.ApprenticeUpdateCreatedMessageProcessorTests
+{
+    public class WhenAApprenticeshipUpdateCreatedMessageIsProcessed
+    {
+        private Mock<IMessageSubscriberFactory> _subscriptionFactory;
+        private Mock<IMessageSubscriber<ApprenticeshipUpdateCreated>> _subscriber;
+        private ApprenticeshipUpdateCreated _messageContent;
+        private Mock<IMessage<ApprenticeshipUpdateCreated>> _mockMessage;
+        private Mock<IMediator> _mediator;
+        private CancellationTokenSource _tokenSource;
+        private ApprenticeUpdateCreatedMessageProcessor _processor;
+
+        [SetUp]
+        public void Arrange()
```

Comment:
Can some of this be moved to a base class or helper or something?  It looks like a lot of it will be common across all handlers.